gitdb/util.py

```diff
@@ -151,7 +151,7 @@ def allocate_memory(size):
 
     try:
         return mmap.mmap(-1, size)  # read-write by default
-    except EnvironmentError:
```

Comment:
I assume that these errors don't exist anymore so there is no other way.
In `GitPython`, I would probably rather be on the side of caution, but here it's probably OK to change the kind of exception to raise. Maybe they are also sub-classing each other so client code wouldn't break because of this.

Reply:
`EnvironmentError` is still there but has been merged into `OSError` since Python 3.3 (released in 2012):

> _Changed in version 3.3:_ [`EnvironmentError`](https://docs.python.org/3/library/exceptions.html#EnvironmentError), [`IOError`](https://docs.python.org/3/library/exceptions.html#IOError), [`WindowsError`](https://docs.python.org/3/library/exceptions.html#WindowsError), [`socket.error`](https://docs.python.org/3/library/socket.html#socket.error), [`select.error`](https://docs.python.org/3/library/select.html#select.error) and `mmap.error` have been merged into [`OSError`](https://docs.python.org/3/library/exceptions.html#OSError), and the constructor may return a subclass.

https://docs.python.org/3/library/exceptions.html#OSError

There are currently no plans to deprecate/remove `EnvironmentError`, but it's cleaner to use `OSError` everywhere.

Reply:
Thanks for sharing your insights, it's much appreciated.

Reply:
And thank you for maintaining GitPython and co!